web/src/server_events_dispatch.js

```diff
@@ -546,20 +546,22 @@ export function dispatch_normal_event(event) {
                         const is_narrowed_to_stream = narrow_state.is_for_stream_id(
                             stream.stream_id,
                         );
+                        stream_data.delete_sub(stream.stream_id);
                         if (is_narrowed_to_stream) {
                             message_lists.current.update_trailing_bookend();
                         }
-                        stream_data.delete_sub(stream.stream_id);
                         stream_settings_ui.remove_stream(stream.stream_id);
+                        message_view_header.maybe_rerender_title_area_for_stream(stream);
```

Comment:
I think this line should be added 2 commits later, when you make the title area rendering depend on whether the stream is archived?

Reply:
I think it will be better to have this whole commit later.